### ISSUE TITLE
Wrong Generic generation with modules

### ISSUE THREAD

**Issue by colderlord** [bridgedotnet/Bridge#3932]
I tried to do an example in Deck, but it's hard because deck can't run modules...

### Steps To Reproduce

https://deck.net/8e001abad6a8d96dfe59e4f6a4e6e81e

Here is my code:

### Module1 ("TestFramework")

```csharp
using System;

public class ComponentTest<TContext, TController> : TestBase<IComponentTestModel<TContext, TController>>
{
    public override TestBuilder<IComponentTestModel<TContext, TController>> InitBuilder()
    {
        return new ComponentTestBuilder<TContext, TController>();
    }
}

public abstract class TestBase<T>
{
    protected TestBuilder<T> builder;

    public abstract TestBuilder<T> InitBuilder();

    public virtual ITestBuilderWhen<T> When(string description, Action<T> action)
    {
        builder = InitBuilder();

        return builder.When(description, action);
    }
}

public interface ITestBuilderWhen<T>
{
    ITestBuilderWhen<T> When(string description, Action<T> action);
    ITestBuilderWhen<T> When(string description, Action action);
}

public interface IComponentTestModel<TContext, TController>
{
    TContext Context { get; }

    TController Controller { get; }
}

public class TestBuilder<T> : ITestBuilderWhen<T>
{
    public T Model { get; set; }

    public ITestBuilderWhen<T> When(string description, Action<T> action)
    {
        return this;
    }

    public ITestBuilderWhen<T> When(string description, Action action)
    {
        return this;
    }

    private Action WrapAction(Action<T> action) => () => action.Invoke(Model);
}

public class ComponentTestBuilder<TContext, TController> : TestBuilder<IComponentTestModel<TContext, TController>>
{
}
```

### Module2 ("MyModule") executor

```csharp
public class App
{
    public static void Main()
    {
        var abc = new ComponentTests();
        abc.Test1();
    }
}

public class ComponentTests : ComponentTest<Context, Controller>
{
    public void Test1()
    {
        When("111", a => a.Context.Str = "123")
            .When("222", b => b.Controller.Click()); // ExceptionHere!
    }
}

public class Controller
{
    public void Click() { }
}

public class Context
{
    public string Str { get; set; }
}
```

I have an exception on 2nd `When`.

### GeneratedCode

```js
this.When("111", $asm.$.MyModule.ComponentTests.f1).TestFramework$ITestBuilderWhen$1$TestFramework$IComponentTestModel$2$MyModule$Context$MyModule$Controller$When$1("222", $asm.$.MyModule.ComponentTests.f2);
```

### In RunTime

```js
this.When("111", $asm.$.MyModule.ComponentTests.f1).TestFramework$ITestBuilderWhen$1$TestFramework$IComponentTestModel$2$Context$Controller$When$1
```

As you can see in Runtime `Context` generated without module `MyModule`

**Comment by colderlord** [comment]
Hello
Any suggestions?

**Comment by geoffreymcgill** [comment]
Thanks for reporting this issue. We will investigate and do our best to fix in time for the upcoming Bridge 17.8.0 release.

**Comment by geoffreymcgill** [comment]
A fix for this issue has been submitted. After the final review and testing, we anticipate the fix will be included in the upcoming Bridge 17.8.0 release.

**Comment by fabriciomurta** [comment]
Fixed! The fix will make it to the next Bridge release!

Just in case, you can "copypaste-thru" requirejs code like in [this deck](https://deck.net/8072161a368a95f83ae9dbe85d1c6045) to have the sample run.
